packages/fast-components-react-base/src/number-field/number-field.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import { get } from "lodash-es";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    NumberFieldHandledProps,
+    NumberFieldManagedClasses,
+    NumberFieldProps,
+    NumberFieldUnhandledProps,
+} from "./number-field.props";
+import {
+    ManagedClasses,
+    NumberFieldClassNameContract,
+} from "@microsoft/fast-components-class-name-contracts-base";
+import { TextFieldType } from "../text-field/index";
+
+class NumberField extends Foundation<
+    NumberFieldHandledProps,
+    NumberFieldUnhandledProps,
+    {}
+> {
+    public static defaultProps: Partial<NumberFieldProps> = {};
+
+    public static displayName: string = "NumberField";
+
+    protected handledProps: HandledProps<NumberFieldHandledProps> = {
+        managedClasses: void 0,
+        disabled: void 0,
+        min: void 0,
+        max: void 0,
+        name: void 0,
+        placeholder: void 0,
+        step: void 0,
+        readOnly: void 0,
+        required: void 0,
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <input
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                type={TextFieldType.number}
+                {...this.assignAttributes()}
+            />
+        );
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        return super.generateClassNames(get(this.props, "managedClasses.numberField"));
+    }
+
+    /**
+     * Generates attributes based on props
+     */
+    private assignAttributes(): Partial<NumberFieldHandledProps> {
+        const attributes: Partial<NumberFieldHandledProps> = {};
+
+        if (typeof this.props.disabled === "boolean") {
+            attributes.disabled = true;
+        }
+
+        if (typeof this.props.max === "number") {
+            attributes.max = this.props.max;
+        }
+
+        if (typeof this.props.min === "number") {
+            attributes.min = this.props.min;
+        }
+
+        if (typeof this.props.placeholder === "string") {
+            attributes.placeholder = this.props.placeholder;
+        }
+
+        if (typeof this.props.name === "string") {
+            attributes.name = this.props.name;
+        }
+
+        if (typeof this.props.readOnly === "boolean") {
+            attributes.readOnly = true;
+        }
+
+        if (typeof this.props.required === "boolean") {
+            attributes.required = true;
+        }
+
+        if (typeof this.props.step === "number") {
+            attributes.step = this.props.step;
+        }

```

Comment:
I question the need to enumerate these, they should already be specified by the unhandledProps, since this should be typed as an HTMLInputElement instead of HTMLDivElement.

Reply:
`HTMLDivElement` to `HTMLInputElement` sure - I missed that.

> I question the need to enumerate **these**

What are these specifically @janechu ? My thought was to expose the common attributes users would expect with a number field. Or else you get a configuration pane without much configuration.

Reply:
All of the props, the fact that this is typed as an HTMLInputElement means this should not need you to enumerate disabled, step, min, max etc.

Reply:
I think it's fine to leave them in the JSON schema because we want to be able to configure those values in the form generator, but you shouldn't need these in the handledProps